lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -5038,6 +5120,12 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
             require(state.CFG == foundState.CFG,
                     "inconsistent coroutine states entering basic block");
           }
+        } else if (i.maySuspend()) {
```

Comment:
This is not exclusive with the conditions above — e.g. `try_apply` could be suspending.

It's probably more efficient to check `GotAsyncContinuation != nullptr` and only then check `i.maySuspend()`.